testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_custom_conflicts.py

```diff
@@ -0,0 +1,1013 @@
+            db.updateDocument(cbl_db, doc_id=doc_id, data=data)
+
+            # Saving the history of update to CBL doc
+            new_docs_body[doc_id].append(data)
+
+    replicator.wait_until_replicator_idle(repl, sleep_time=12)  # Added sleep time because CCR sleeps for 10 secs
+    total = replicator.getTotal(repl)
+    completed = replicator.getCompleted(repl)
+    replicator.stop(repl)
+    assert total == completed, "total is not equal to completed"
+
+    sg_docs_content = sg_client.get_bulk_docs(sg_url, sg_db, doc_ids, session)[0]
+    cbl_docs = db.getDocuments(cbl_db, doc_ids)
+
+    if replicator_type == "pull":
+        for sg_doc in sg_docs_content:
+            doc_id = sg_doc["_id"]
+            cbl_doc = cbl_docs[doc_id]
+            assert sg_doc["sg_new_update1"] != cbl_doc["sg_new_update1"], "CCR failed to resolve conflict " \
+                                                                          "with delayed local win"
+            assert sg_doc["sg_new_update2"] != cbl_doc["sg_new_update2"], "CCR failed to resolve conflict " \
+                                                                          "with delayed local win"
+            assert sg_doc["sg_new_update3"] != cbl_doc["sg_new_update3"], "CCR failed to resolve conflict " \
+                                                                          "with delayed local win"
+            assert "random" not in cbl_doc, "CCR failed to resolve conflict with delayed local win"
+            assert "cbl_random" not in sg_doc, "CCR failed to resolve conflict with delayed local win"
+            assert new_docs_body[doc_id][1]["update_during_CCR"] == cbl_doc["update_during_CCR"], "CCR failed to " \
+                                                                                                  "resolve conflict " \
+                                                                                                  "with delayed " \
+                                                                                                  "local win"
+    elif replicator_type == "push_pull":
+        for sg_doc in sg_docs_content:
+            doc_id = sg_doc["_id"]
+            cbl_doc = cbl_docs[doc_id]
+            assert sg_doc["sg_new_update1"] == cbl_doc["sg_new_update1"], "CCR failed to resolve conflict " \
+                                                                          "with delayed local win"
+            assert sg_doc["sg_new_update2"] == cbl_doc["sg_new_update2"], "CCR failed to resolve conflict " \
+                                                                          "with delayed local win"
+            assert sg_doc["sg_new_update3"] == cbl_doc["sg_new_update3"], "CCR failed to resolve conflict " \
+                                                                          "with delayed local win"
+            assert "random" not in sg_doc, "CCR failed to resolve conflict with delayed local win"
```

Comment:
sorry , can you point me where this key added?

Reply:
check doc update code